File: car-rent-back/car-rent-back/Models/Car.cs
namespace car_rent_back.Models;

using System.Text.Json.Serialization;

public enum TransmissionType
{
    Manual,
    Automatic,
    SemiAutomatic,
    CVT,
    Robot
}

public enum FuelType
{
    Petrol,
    Diesel,
    Gas,
    Hybrid,
    Electric
}

public class Car
{
    public Guid Id { get; set; }
    public string Brand { get; set; } = null!;
    public string Model { get; set; } = null!;
    public int Year { get; set; }
    public string LicensePlate { get; set; } = null!;
    public string Color { get; set; } = null!;
    public decimal PricePerHour { get; set; }
    public decimal PricePerDay { get; set; }
    public bool IsAvailable { get; set; } = true;
    public string? ImageUrl { get; set; }
    public string? Description { get; set; }
    
    [JsonConverter(typeof(JsonStringEnumConverter))]
    public TransmissionType Transmission { get; set; }
    
    [JsonConverter(typeof(JsonStringEnumConverter))]
    public FuelType FuelType { get; set; }
    
    // Навигационные свойства
    [JsonIgnore]
    public virtual ICollection<Rental> Rentals { get; set; } = new List<Rental>();
}

File: car-rent-vite/src/types/car.ts
type RentalStatus = 'Booked' | 'Active' | 'Completed' | 'Cancelled';

export interface Rental {
  id: string;
  carId: string;
  userId: string;
  startDateTime: string;
  endDateTime: string;
  status: RentalStatus;
  totalCost: number;
}

export interface Car {
  id: string;
  brand: string;
  model: string;
  year: number;
  licensePlate: string;
  color: string;
  pricePerHour: number;
  pricePerDay: number;
  isAvailable: boolean;
  imageUrl?: string;
  description?: string;
  fuelType: string;
  transmission: string;
  rentals?: Rental[];
}
